core/src/main/java/org/apache/calcite/rel/logical/RavenDistinctProject.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.rel.logical;
+
+import org.apache.calcite.plan.Convention;
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.RelCollation;
+import org.apache.calcite.rel.RelCollationTraitDef;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.core.Project;
+import org.apache.calcite.rel.metadata.RelMdCollation;
+import org.apache.calcite.rel.metadata.RelMetadataQuery;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rex.RexNode;
+import org.apache.calcite.rex.RexUtil;
+import org.apache.calcite.sql.validate.SqlValidatorUtil;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+/**
+ * RavenDistinctProject is an extension of Project which represent distinct by a boolean value.
+ */
+
+
+public class RavenDistinctProject extends Project {
+
+  boolean distinct;
```

Comment:
Hi @praveendatametica 
I dont think this is the right way to model a distinct. Please discuss.

Reply:
Sure , will discuss the same internally

Reply:
We are not adding any logic here just moving RavenDistinctProject from mig to calcite.
Earlier we used to have distinct flag in project but in new update this flag is removed , so to figure out distinct rel node we have moved this class.

I know this is not the best approach , but for the time being can we proceed with merge , and parallelly  discuss the right way to figure out distinct in calcite

Reply:
I dont think there was ever a flag in the Project for the distinct boolean. The fields used in the project were used to create a LogicalAggregate whose group by fields were the fields of the Project. You can see this code in RelBuilder.

We already have an approach to identify a distinct i.e look to see if the project fields are the same as the aggregate fields. We should proceed with it.

Reply:
Hi Abbas, so converting the LogicalAgg -> LogicalProject thing back to a Distinct keyword will be done(already exists or we can do something to make it so) OR are you suggesting we should  generate `select id, name from t1 group by id, name ` to achieve DISTINCT behaviour ? 

Also, there was a isDistinct() behaviour in LogicalProject before the big update in OUR calcite version which this flag was overriding. This RavenDistinctProject class was added in that pre update phase. Now the API got removed but this class still exists.
```
@SuppressWarnings("deprecation")
  public boolean isDistinct() {
    final RelMetadataQuery mq = RelMetadataQuery.instance();
    return Boolean.TRUE.equals(mq.areRowsUnique(this));
  }
```